executor/src/witgen/generator.rs

```diff
@@ -204,16 +192,7 @@ where
         self.next = values.iter().cloned().map(Some).collect();
 
         for identity in self.identities {
-            let result = match identity.kind {
-                IdentityKind::Polynomial => {
-                    self.process_polynomial_identity(identity.left.selector.as_ref().unwrap())
-                }
-                IdentityKind::Plookup | IdentityKind::Permutation => self.process_plookup(identity),
-                kind => {
-                    unimplemented!("Identity of kind {kind:?} is not supported in the executor")
-                }
-            };
-            if result.is_err() {
+            if self.process_identity(identity).is_err() {
```

Comment:
The error gets lost here. Is it not needed?

Reply:
This is the loop part. If we fail with the proposed values, we try the regular deduction method that could succeed or at least provide a better error.